Parser/GenericParser/InstrumentParsers/StaroddiParser.m

```diff
@@ -0,0 +1,185 @@
+sdata.mode = mode;
+[sdata.file_encoding, sdata.file_machineformat] = detectEncoding(filename);
+
+fid = fopen(filename, 'r', sdata.file_machineformat, sdata.file_encoding);
+[sdata.header_info, sdata.header_content, sdata.header_lines] = HeaderParser(fid, sdata.header_def_rules, sdata.header_oper_rules);
+[sdata.rawdata, sdata.procdata] = DataParser(fid, sdata);
+data = mapDataNames(sdata.procdata, sdata.name_map_rules, sdata.header_info);
+sdata.data = data;
+fclose(fid);
+
+% map variable names
+hvmap = containers.Map();
+hvmap('Temperature') = 'TEMP';
+hvmap('Pressure') = 'PRES';
+hvmap('Depth') = 'DEPTH';
+hvmap('Salinity') = 'PSAL';
+hvmap('Conductivity') = 'CNDC';
+hvmap('SoundVelocity') = 'SOUND_VEL';
+hvmap('Roll') = 'ROLL';
+hvmap('Pitch') = 'PITCH';
+
+%map dimension names
+hdmap = containers.Map();
+hdmap('Temperature') = 1;
+hdmap('Pressure') = 1;
+hdmap('Depth') = 1;
+hdmap('Salinity') = 1;
+hdmap('Conductivity') = 1;
+hdmap('SoundVelocity') = 1;
+hdmap('Roll') = 1;
+hdmap('Pitch') = 1;
+
+sample_data = struct;
+sample_data.toolbox_input_file = filename;
+sample_data.meta.header = sdata.header_info;
+sample_data.meta.instrument_make = 'Star ODDI';
+sample_data.meta.instrument_model = sdata.header_info.recorder.instrument_name;
+sample_data.meta.instrument_serial_no = sdata.header_info.recorder.serial_number;
+sample_data.meta.instrument_sample_interval = median(diff(data.('TIME') * 24 * 3600));
+sample_data.meta.featureType = mode;
+
+[coordinates, sample_data.dimensions, sample_data.variables] = loadTimeSeriesSampleTemplate(data.('TIME'));
+converted_variables = convertVariables(data, hvmap, hdmap, coordinates);
+
+sample_data.variables = [sample_data.variables, converted_variables];
+
+is_degF_mini = isfield(sdata.header_info, 'channel_index_1') && isfield(sdata.header_info,'channel_info_1') && strcmp(sdata.header_info.channel_info_1.channel_units, '°F');
+is_degF_dst = isfield(sdata.header_info, 'axis_index_0') && isfield(sdata.header_info,'axis_info_0') && strcmp(sdata.header_info.axis_info_0.axis_units, '°F');
+is_temp_fahrenheit = is_degF_mini || is_degF_dst;
+
+if is_temp_fahrenheit
+    for k = 1:length(sample_data.variables)
+        vname = sample_data.variables{k}.name;
+        is_temp = strcmp(vname, 'TEMP') || contains(vname, 'TEMP_');
+
+        if is_temp
+            sample_data.variables{k}.data = toCelsius(sample_data.variables{k}.data,'fahrenheit');
+            sample_data.variables{k}.comment = 'Originaly expressed in Fahrenheit.';
+        end
+
+    end
+
+end
+
+is_temp_corrected = isfield(sdata.header_info, 'no_temperature_correction') && ~sdata.header_info.no_temperature_correction;
+is_pres_corrected = isfield(sdata.header_info, 'pressure_offset_correction') && sdata.header_info.pressure_offset_correction;
+is_salt_corrected = (is_temp_corrected || is_pres_corrected) && getVar(sample_data.variables, 'PSAL');
+
+if is_temp_corrected
+    k = getVar(sample_data.variables, 'TEMP_2');
```

Comment:
Why not using resolveIMOSName to generate this new TEMP_2 , unless I'm missing the point?
same for depth_rel below and psal_rel ...

Reply:
explicit is better than implicit in this case - there should be only one or two temperatures in those instruments.